src/kafka.js

```diff
@@ -0,0 +1,53 @@
+'use strict'
+
+const {KafkaProducer, KafkaConsumer} = require('./kafkaUtils')
+const {port} = require('./config').getConfig()
+const logger = require('./logger')
+
+let kafkaConsumer, kafkaProducer
+
+const initiateKafkaClient = async (endpoints = []) => {
+  kafkaConsumer = new KafkaConsumer()
+  kafkaProducer = new KafkaProducer()
+
+  for (let i = 0; i < endpoints.length; i++) {
+    const endpoint = endpoints[i]
+
+    await subscribeTopicToConsumer(endpoint)
+  }
+}
+
+const subscribeTopicToConsumer = async endpoint => {
+  if (endpoint.kafkaConsumerTopic) {
+    await kafkaConsumer.subscribe(endpoint.kafkaConsumerTopic, {
```

Comment:
We need to either wrap this in a try/catch, or make sure that every place that calls it handles the error path (so make sure we call next(err) in the mongo models so we don't wait forever).

Currently if you have a kafka consumer endpoint saved but kafak is not running, if you try and start the project it will die ("uncaught KafkaJSConnectionClosedError: Closed connection").

Reply:
This will allow for other endpoints to still run and does not create a hard dependency on kafka being up and running for the mapping mediator to function. The only issue now is the consumer will never subscribe if we start the mapping mediator if kafka was down at the same time. It may be a good idea to catch the ConnectionClosedError itself and then write polling logic to try and reconnect once kafka is up and running again.

Reply:
Addressed in https://github.com/jembi/openhim-mediator-mapping/pull/205/commits/55e9fa320987315eb714daea597f6e93775da7c0